Config tests should ignore filesystem

When developing locally, it's convenient to put a `pbs.yml` in the project root with your config values.

Currently, there's a "unit" test which fails based on the contents in this file:

```
--- FAIL: TestViperInit (0.00s)
	pbs_light_test.go:685: Viper error: external_url expected to be http://localhost:8000, found {other-value}
FAIL
FAIL	github.com/prebid/prebid-server	0.069s
```

We should fix these tests so that they don't check the files on the filesystem.

--- a/config/config.go
+++ b/config/config.go
@@ -237,7 +237,7 @@ func (cfg *Configuration) GetCachedAssetURL(uuid string) string {
 // Set the default config values for the viper object we are using.
 func SetupViper(v *viper.Viper, filename string) {
 	if filename != "" {
-		v.SetConfigName(name)
+		v.SetConfigName(filename)
 		v.AddConfigPath(".")
 		v.AddConfigPath("/etc/config")
 	}

--- a/config/config.go
+++ b/config/config.go
@@ -235,8 +235,8 @@ func (cfg *Configuration) GetCachedAssetURL(uuid string) string {
 }
 
 // Set the default config values for the viper object we are using.
-func SetupViper(v *viper.Viper) {
-	v.SetConfigName("pbs")
+func SetupViper(v *viper.Viper, name string) {
+	v.SetConfigName(name)
 	v.AddConfigPath(".")
 	v.AddConfigPath("/etc/config")
 

--- a/config/config.go
+++ b/config/config.go
@@ -235,11 +235,12 @@ func (cfg *Configuration) GetCachedAssetURL(uuid string) string {
 }
 
 // Set the default config values for the viper object we are using.
-func SetupViper(v *viper.Viper, name string) {
-	v.SetConfigName(name)
-	v.AddConfigPath(".")
-	v.AddConfigPath("/etc/config")
-
+func SetupViper(v *viper.Viper, filename string) {
+	if filename != "" {
+		v.SetConfigName(name)
+		v.AddConfigPath(".")
+		v.AddConfigPath("/etc/config")
+	}
 	// Fixes #475: Some defaults will be set just so they are accessable via environment variables
 	// (basically so viper knows they exist)
 	v.SetDefault("external_url", "http://localhost:8000")

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -13,7 +13,7 @@ import (
 
 func TestDefaults(t *testing.T) {
 	v := viper.New()
-	SetupViper(v)
+	SetupViper(v, "abacadaeaf")
 	cfg, err := New(v)
 	if err != nil {
 		t.Error(err.Error())
@@ -104,7 +104,7 @@ func cmpBools(t *testing.T, key string, a bool, b bool) {
 
 func TestFullConfig(t *testing.T) {
 	v := viper.New()
-	SetupViper(v)
+	SetupViper(v, "abacadaeaf")
 	v.SetConfigType("yaml")
 	v.ReadConfig(bytes.NewBuffer(fullConfig))
 	cfg, err := New(v)

--- a/config/config_test.go
+++ b/config/config_test.go
@@ -13,7 +13,7 @@ import (
 
 func TestDefaults(t *testing.T) {
 	v := viper.New()
-	SetupViper(v, "abacadaeaf")
+	SetupViper(v, "")
 	cfg, err := New(v)
 	if err != nil {
 		t.Error(err.Error())
@@ -104,7 +104,7 @@ func cmpBools(t *testing.T, key string, a bool, b bool) {
 
 func TestFullConfig(t *testing.T) {
 	v := viper.New()
-	SetupViper(v, "abacadaeaf")
+	SetupViper(v, "")
 	v.SetConfigType("yaml")
 	v.ReadConfig(bytes.NewBuffer(fullConfig))
 	cfg, err := New(v)

--- a/pbs_light.go
+++ b/pbs_light.go
@@ -634,7 +634,7 @@ func init() {
 
 func main() {
 	v := viper.New()
-	config.SetupViper(v)
+	config.SetupViper(v, "pbs")
 	cfg, err := config.New(v)
 	if err != nil {
 		glog.Fatalf("Configuration could not be loaded or did not pass validation: %v", err)

--- a/pbs_light_test.go
+++ b/pbs_light_test.go
@@ -350,7 +350,7 @@ func TestCacheVideoOnly(t *testing.T) {
 	ctx := context.TODO()
 	w := httptest.NewRecorder()
 	v := viper.New()
-	config.SetupViper(v)
+	config.SetupViper(v, "abacadaeaf")
 	cfg, err := config.New(v)
 	if err != nil {
 		t.Fatal(err.Error())
@@ -655,14 +655,14 @@ func (validator *testValidator) Schema(name openrtb_ext.BidderName) string {
 // Test the viper setup
 func TestViperInit(t *testing.T) {
 	v := viper.New()
-	config.SetupViper(v)
+	config.SetupViper(v, "abacadaeaf")
 	compareStrings(t, "Viper error: external_url expected to be %s, found %s", "http://localhost:8000", v.Get("external_url").(string))
 	compareStrings(t, "Viper error: adapters.pulsepoint.endpoint expected to be %s, found %s", "http://bid.contextweb.com/header/s/ortb/prebid-s2s", v.Get("adapters.pulsepoint.endpoint").(string))
 }
 
 func TestViperEnv(t *testing.T) {
 	v := viper.New()
-	config.SetupViper(v)
+	config.SetupViper(v, "abacadaeaf")
 	port := forceEnv(t, "PBS_PORT", "7777")
 	defer port()
 

--- a/pbs_light_test.go
+++ b/pbs_light_test.go
@@ -350,7 +350,7 @@ func TestCacheVideoOnly(t *testing.T) {
 	ctx := context.TODO()
 	w := httptest.NewRecorder()
 	v := viper.New()
-	config.SetupViper(v, "abacadaeaf")
+	config.SetupViper(v, "")
 	cfg, err := config.New(v)
 	if err != nil {
 		t.Fatal(err.Error())
@@ -655,14 +655,14 @@ func (validator *testValidator) Schema(name openrtb_ext.BidderName) string {
 // Test the viper setup
 func TestViperInit(t *testing.T) {
 	v := viper.New()
-	config.SetupViper(v, "abacadaeaf")
+	config.SetupViper(v, "")
 	compareStrings(t, "Viper error: external_url expected to be %s, found %s", "http://localhost:8000", v.Get("external_url").(string))
 	compareStrings(t, "Viper error: adapters.pulsepoint.endpoint expected to be %s, found %s", "http://bid.contextweb.com/header/s/ortb/prebid-s2s", v.Get("adapters.pulsepoint.endpoint").(string))
 }
 
 func TestViperEnv(t *testing.T) {
 	v := viper.New()
-	config.SetupViper(v, "abacadaeaf")
+	config.SetupViper(v, "")
 	port := forceEnv(t, "PBS_PORT", "7777")
 	defer port()
 

